### PR TITLE
Code changes for "Handling errors in conversation events"

### PR DESCRIPTION
**Pull request by Keshav-MSFT**

### CODE REVIEW

File: samples/app-checkin-location/nodejs/index.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ adapter.onTurnError = async (context, error) => {
     //       application insights. See https://aka.ms/bottelemetry for telemetry 
     //       configuration instructions.
     console.error(`\n [onTurnError] unhandled error: ${error}`);
+    await context.sendTraceActivity(`Sorry, it looks like something went wrong. Exception Caught: ${error}`);
```
Comment (Pawan-MSFT) [inline samples/app-checkin-location/nodejs/index.js:37]: This is not needed, Why we are using sendTraceActivity? because below line same is available.

Comment (Pawan-MSFT) [reply samples/app-checkin-location/nodejs/index.js:37]: after below line need to add that code in commented mode.
 await context.sendActivity('To continue to run this bot, please fix the bot source code.');

File: samples/bot-conversation/csharp/AdapterWithErrorHandler.cs [hunk 2]
```diff
@@ -21,8 +21,8 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<IBotFram
                 // to add telemetry capture to your bot.
                 logger.LogError($"Exception caught : {exception.Message}");
 
-                // Send a catch-all apology to the user.
-                await turnContext.SendActivityAsync("Sorry, it looks like something went wrong.");
+                // Uncomment this for local debugging.
+                await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
```
Comment (Pawan-MSFT) [inline samples/bot-conversation/csharp/AdapterWithErrorHandler.cs:25]: As discussed, this should be commented.

File: samples/bot-initiate-thread-in-channel/csharp/AdapterWithErrorHandler.cs [hunk 3]
```diff
@@ -21,8 +21,8 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<IBotFram
                 // to add telemetry capture to your bot.
                 logger.LogError($"Exception caught : {exception.Message}");
 
-                // Send a catch-all apology to the user.
-                await turnContext.SendActivityAsync("Sorry, it looks like something went wrong.");
+                // Uncomment this for local debugging.
+                await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
```
Comment (Pawan-MSFT) [inline samples/bot-initiate-thread-in-channel/csharp/AdapterWithErrorHandler.cs:25]: As discussed, this should be commented.

File: samples/bot-message-reaction/csharp/AdapterWithErrorHandler.cs [hunk 4]
```diff
@@ -20,8 +20,8 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<IBotFram
                 // to add telemetry capture to your bot.
                 logger.LogError($"Exception caught : {exception.Message}");
 
-                // Send a catch-all apology to the user.
-                await turnContext.SendActivityAsync("Sorry, it looks like something went wrong.");
+                // Uncomment this for local debugging.
+                await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
```
Comment (Pawan-MSFT) [inline samples/bot-message-reaction/csharp/AdapterWithErrorHandler.cs:24]: As discussed, this should be commented.

File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/Bot/BotHttpAdapter.cs [hunk 5]
```diff
@@ -20,11 +20,11 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<IBotFram
                 // to add telemetry capture to your bot.
                 logger.LogError($"Exception caught : {exception.Message}");
 
-                // Send a catch-all apology to the user.
-                await turnContext.SendActivityAsync("Sorry, it looks like something went wrong.");
+            // Uncomment this for local debugging.
+            await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
```
Comment (Pawan-MSFT) [inline samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/Bot/BotHttpAdapter.cs:24]: As discussed, this should be commented.

File: samples/bot-task-module/csharp/AdapterWithErrorHandler.cs [hunk 6]
```diff
@@ -20,8 +20,8 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<IBotFram
                 // to add telemetry capture to your bot.
                 logger.LogError($"Exception caught : {exception.Message}");
 
-                // Send a catch-all apology to the user.
-                await turnContext.SendActivityAsync("Sorry, it looks like something went wrong.");
+                // Uncomment this for local debugging.
+                await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
```
Comment (Pawan-MSFT) [inline samples/bot-task-module/csharp/AdapterWithErrorHandler.cs:24]: As discussed, this should be commented.

**Review by Pawan-MSFT — COMMENTED** [review]
Please make sure we've covered all the apps code.

File: samples/app-checkin-location/nodejs/index.js (hunk 1 above)
Comment (Keshav-MSFT) [reply samples/app-checkin-location/nodejs/index.js:37]: resolved

File: samples/bot-conversation/csharp/AdapterWithErrorHandler.cs (hunk 2 above)
Comment (Keshav-MSFT) [reply samples/bot-conversation/csharp/AdapterWithErrorHandler.cs:25]: resolved

File: samples/bot-initiate-thread-in-channel/csharp/AdapterWithErrorHandler.cs (hunk 3 above)
Comment (Keshav-MSFT) [reply samples/bot-initiate-thread-in-channel/csharp/AdapterWithErrorHandler.cs:25]: resolved

File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/Bot/BotHttpAdapter.cs (hunk 5 above)
Comment (Keshav-MSFT) [reply samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/Bot/BotHttpAdapter.cs:24]: resolved

File: samples/bot-task-module/csharp/AdapterWithErrorHandler.cs (hunk 6 above)
Comment (Keshav-MSFT) [reply samples/bot-task-module/csharp/AdapterWithErrorHandler.cs:24]: resolved

File: samples/app-checkin-location/nodejs/index.js [hunk 7]
```diff
@@ -46,6 +47,8 @@ adapter.onTurnError = async (context, error) => {
     // Send a message to the user
     await context.sendActivity('The bot encountered an error or bug.');
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:48]: Remove these lines from all of your files.

File: samples/app-checkin-location/nodejs/index.js [hunk 8]
```diff
@@ -34,6 +34,7 @@ adapter.onTurnError = async (context, error) => {
     //       application insights. See https://aka.ms/bottelemetry for telemetry 
     //       configuration instructions.
     console.error(`\n [onTurnError] unhandled error: ${error}`);
+
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:37]: Remove extra empty line space.

File: samples/app-checkin-location/nodejs/index.js [hunk 9]
```diff
@@ -34,6 +34,7 @@ adapter.onTurnError = async (context, error) => {
     //       application insights. See https://aka.ms/bottelemetry for telemetry 
     //       configuration instructions.
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:35]: Add only one space and remove extra spaces.

File: samples/app-checkin-location/nodejs/index.js [hunk 10]
```diff
@@ -46,6 +47,8 @@ adapter.onTurnError = async (context, error) => {
     // Send a message to the user
     await context.sendActivity('The bot encountered an error or bug.');
     await context.sendActivity('To continue to run this bot, please fix the bot source code.');
+     // Uncomment this for local debugging.
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:50]: // Uncomment below commented line for local debugging.

File: samples/app-checkin-location/nodejs/index.js [hunk 11]
```diff
@@ -46,6 +47,8 @@ adapter.onTurnError = async (context, error) => {
...
+    //  await context.sendActivity(`Sorry, it looks like something went wrong. Exception Caught: ${error}`);
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:51]: // await context.sendActivity(`Sorry, it looks like something went wrong. Exception Caught: ${error}`);

File: samples/app-checkin-location/nodejs/index.js [hunk 12]
```diff
@@ -46,6 +47,8 @@ adapter.onTurnError = async (context, error) => {
...
 };
 
 // Create the bot that will handle incoming messages.
```
Comment (ChetanSharma-msft) [inline samples/app-checkin-location/nodejs/index.js:54]: Add a note in all the Readme files in Setup section like below:
**Note:** "If you are facing any issue in your app, please uncomment [this ](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-keshsharma/updatedcode/samples/app-checkin-location/nodejs/index.js#L51) line and put your debugger for local debug."

File: samples/app-complete-auth/nodejs/index.js [hunk 13]
```diff
@@ -63,7 +63,8 @@ adapter.onTurnError = async (context, error) => {
     'https://www.botframework.com/schemas/error',
     'TurnError'
   );
-
```
Comment (ChetanSharma-msft) [inline samples/app-complete-auth/nodejs/index.js:66]: Add empty line

File: samples/app-complete-auth/csharp/AppCompleteAuth/AdapterWithErrorHandler.cs [hunk 14]
```diff
@@ -29,6 +29,8 @@ public AdapterWithErrorHandler(BotFrameworkAuthentication auth, ILogger<BotFrame
 
                 // Send a trace activity, which will be displayed in the Bot Framework Emulator
                 await turnContext.TraceActivityAsync("OnTurnError Trace", exception.Message, "https://www.botframework.com/schemas/error", "TurnError");
+                // Uncomment this for local debugging.
```
Comment (ChetanSharma-msft) [inline samples/app-complete-auth/csharp/AppCompleteAuth/AdapterWithErrorHandler.cs:32]: Add empty line

File: samples/app-complete-auth/nodejs/index.js [hunk 15]
```diff
@@ -63,7 +63,8 @@ adapter.onTurnError = async (context, error) => {
...
+   // Uncomment this for local debugging.
+  //await context.sendActivity(`Sorry, it looks like something went wrong. Exception Caught: ${error}`);
```
Comment (ChetanSharma-msft) [inline samples/app-complete-auth/nodejs/index.js:67]: // await context.sendActivity(`Sorry, it looks like something went wrong. Exception Caught: ${error}`);

File: samples/app-complete-sample/csharp/AdapterWithErrorHandler.cs [hunk 16]
```diff
@@ -19,7 +22,8 @@ public AdapterWithErrorHandler(IConfiguration configuration, ILogger<IBotFramewo
                 // Send a message to the user
                 await turnContext.SendActivityAsync("The bot encountered an error or bug.");
                 await turnContext.SendActivityAsync("To continue to run this bot, please fix the bot source code.");
```
Comment (ChetanSharma-msft) [inline samples/app-complete-sample/csharp/AdapterWithErrorHandler.cs:24]: Remove these lines.

File: samples/app-complete-sample/csharp/AdapterWithErrorHandler.cs [hunk 17]
```diff
@@ -19,7 +22,8 @@ public AdapterWithErrorHandler(IConfiguration configuration, ILogger<IBotFramewo
...
-
+                // Uncomment this for local debugging.
+                // await turnContext.SendActivityAsync($"Sorry, it looks like something went wrong. Exception Caught: {exception.Message}");
                 if (conversationState != null)
```
Comment (ChetanSharma-msft) [inline samples/app-complete-sample/csharp/AdapterWithErrorHandler.cs:27]: Add empty extra line.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: samples/app-checkin-location/nodejs/index.js (hunk 10 above)
Comment (Keshav-MSFT) [reply samples/app-checkin-location/nodejs/index.js]: resolved

File: samples/bot-proactive-messaging-teamsfx/README.md [hunk 18]
```diff
@@ -23,6 +23,9 @@ You can find the Teams app manifest in `templates/appPackage` folder. The folder
 
 Both files contain template arguments with `{...}` statements which will be replaced at build time. You can hover the prepoerties to see the actual value or preview the manifest. You may add any extra properties or permissions you require to this file. See the [schema reference](https://docs.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema) for more information.
 
+Note:- If you are facing any issue in your app,  [please uncomment this line](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-proactive-messaging-teamsfx/bot/index.js#L28) and put your debugger for local debug.
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-proactive-messaging-teamsfx/README.md:26]: This should be properly updated in all the files.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

File: samples/app-checkin-location/nodejs/Readme.md [hunk 19]
```diff
@@ -69,6 +69,7 @@ This sample shows feature where user can checkin with current location and view
     ```bash
     npm start
     ```
+Note:-   If you are facing any issue in your app,  [please uncomment this line]https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/app-checkin-location/nodejs/index.js#L48) and put your debugger for local debug.
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/app-checkin-location/nodejs/Readme.md:72]: **Note:** If you are facing any issue in your app, [please uncomment this line]https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/app-checkin-location/nodejs/index.js#L48) and put your debugger for local debug.

[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/app-checkin-location/nodejs/Readme.md:72]: ( is missing in your URL

File: samples/app-complete-auth/nodejs/README.md [hunk 20]
```diff
@@ -158,6 +158,7 @@ Using this C# sample, you can check authenticate in bot,tab and messaging extent
    - Select Add in the pop-up dialog box. Your app is uploaded to Teams.
 
 - Note: To test facebook auth flow please setup the sample locally as due to limitations from facebook you cannot test the facebook auth flow in the deployed version of app.
+- If you are facing any issue in your app,  [please uncomment this line](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/app-complete-auth/nodejs/index.js#L68) and put your debugger for local debug.
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/app-complete-auth/nodejs/README.md:161]: **Note:**

**Review by Pawan-MSFT — COMMENTED** [review]
please fix

File: samples/bot-archive-groupchat-messages/csharp/README.md [hunk 21]
```diff
@@ -126,6 +126,8 @@ This feature shown in this sample is currently available in Public Developer Pre
 	   - From the lower left corner, choose Upload a custom App
 	   - Go to your project directory, the ./manifest folder, select the zip folder, and choose Open.
 	   - Select Add in the pop-up dialog box. Your app is uploaded to Teams.
+
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-keshsharma/updatedcode/samples/bot-archive-groupchat-messages/csharp/FetchGroupChatMessages/AdapterWithErrorHandler.cs#L23) line and put your debugger for local debug.
```
Comment (ChetanSharma-msft) [inline samples/bot-archive-groupchat-messages/csharp/README.md:130]: Why you have added your branch link?

File: samples/bot-join-team-using-qr-code/nodejs/Readme.md [hunk 22]
```diff
@@ -99,6 +99,8 @@ b) Navigate to **API Permissions**, and make sure to add the follow permissions:
 
 > IMPORTANT: The manifest file in this app adds "token.botframework.com" to the list of `validDomains`. This must be included in any bot that uses the Bot Framework OAuth flow.
 
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-keshsharma/updatedcode/samples/bot-join-team-using-qr-code/nodejs/index.js#L50) line and put your debugger for local debug.
```
Comment (ChetanSharma-msft) [inline samples/bot-join-team-using-qr-code/nodejs/Readme.md:102]: Why you have added your branch sample link?

File: samples/bot-release-management/csharp/README.md [hunk 23]
```diff
@@ -151,6 +151,8 @@ Click on Add Permissions to commit your changes.
     - **Upload** the `manifest.zip` to Teams (In Teams Apps/Manage your apps click "Upload an app to your org's app catalog". Browse to and Open the .zip file. At the next dialog, click the Add button.)
     - Add the app to personal/team/groupChat scope (Supported scopes)
 
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-keshsharma/updatedcode/samples/bot-release-management/csharp/ReleaseManagement/AdapterWithErrorHandler.cs#L22) line and put your debugger for local debug.
```
Comment (ChetanSharma-msft) [inline samples/bot-release-management/csharp/README.md:154]: Don't add your branch sample link

File: samples/bot-suggested-actions/csharp/README.md [hunk 24]
```diff
@@ -88,6 +88,7 @@ This sample shows the feature where user can send suggested actions using bot.
    - Go to your project directory, the ./AppPackage folder, select the zip folder, and choose Open.
    - Select Add in the pop-up dialog box. Your app is uploaded to Teams.
 
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/v-keshsharma/updatedcode/samples/bot-suggested-actions/csharp/SuggestedActions/AdapterWithErrorHandler.cs#L28) line and put your debugger for local debug.
```
Comment (ChetanSharma-msft) [inline samples/bot-suggested-actions/csharp/README.md:91]: Don't add your branch sample link

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approved.

**Comment by ChetanSharma-msft** [comment]
Execute build pipeline:

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**